packages/protocol/contracts/governance/Governance.sol

```diff
@@ -1472,14 +1472,20 @@ contract Governance is
 
     uint256 maxUsed = 0;
     for (uint256 index = 0; index < dequeued.length; index = index.add(1)) {
-      Proposals.Proposal storage proposal = proposals[dequeued[index]];
+      uint256 proposalId = dequeued[index];
+      Proposals.Proposal storage proposal = proposals[proposalId];
       bool isVotingReferendum = (getProposalDequeuedStage(proposal) == Proposals.Stage.Referendum);
 
       if (!isVotingReferendum) {
         continue;
       }
 
       VoteRecord storage voteRecord = voter.referendumVotes[index];
+      // skip if vote record is not for this proposal
+      if (voteRecord.proposalId != proposalId) {
+        continue;
```

Comment:
wondering if this may be a good opportunity to delete that `voteRecord `. Ditto bellow.

Reply:
I remove it lowe, it is not possible to do it in this function since it is view function